### DISCUSSION TITLE
Set color of nickname from spec

### DISCUSSION THREAD

**Discussion by abreiter** [wxyz-abcd/node-haxball#64] [Ideas]
So, I have a 24-hour room at haxball, I have a script, the room is Space Bounce Brasil, you must have seen it.
Well, sorry to create this as an "issue" but I didn't know how to contact you.

Come on, I would need it to be possible to edit the color of a nick/playerName in the spectator list, as I have specific treatment for AFK players.

I would like these AFK players to have a certain color, so that players who choose teammates can differentiate them, since my script does not allow choosing a player who is AFK.

Can you help me ?

Suddenly create a NickColor type property on the "Player" object.

Thanks.

God bless you.

**Comment by wxyz-abcd** [comment]
Do you need everyone to see the colors?

- If so, it will not be possible unless you create a custom haxball client using this API and create/send customEvent so that it will be received by your custom clients and they will update their guis automatically. And you can create the NickColor property by default on all Player objects inside room.onPlayerObjectCreated callback.

- Otherwise, you do not need this API at all. You can create an extension or something like that to modify the colors of any player on any browser.

We have a discord channel. You can check the announcements page here. :)

**Reply by abreiter** [reply]
Yes, i need everyone to see that color.
For example, i am AFK, i would like everyone see my nickname color as gray.
Is it so hard to implement ?
I am a nodejs begginer, pro in delphi, lol....

**Reply by wxyz-abcd** [reply]
Unfortunately both the host and client logic are located inside the client side of Haxball. That's why it's hard to modify. In fact, it would be much harder if the host side was located inside webserver, but still, the clients have to be the same for your buddies to be able to see the color effect visually. Implementing it is in fact not hard, but you need a custom web hosting and invite your friends to your own website, that is another story. And no, you don't need node.js at all, you can do everything directly inside the browser. The api works probably on all environments.
